apps/src/gamelab/GameLabP5.js

```diff
@@ -608,6 +608,41 @@ GameLabP5.prototype.preloadBackgrounds = function() {
   );
 };
 
+GameLabP5.prototype.preloadSpriteImages = function(animationList) {
+  if (!this.preloadedSprites) {
+    this.preloadedSprites = {};
+  }
+  this.preloadSpriteImages_ = Promise.all(
+    animationList.orderedKeys.map(key => {
+      const props = animationList.propsByKey[key];
+      if (
+        this.preloadedSprites[props.name] &&
+        this.preloadedSprites[props.name].dataURI === props.dataURI
+      ) {
```

Comment:
Should we be doing the same thing for backgrounds?

Reply:
Students cannot add or edit backgrounds, so preloading the set list on initial page load is sufficient. For sprites, students can add new sprites from the library and/or edit existing sprites at any time, so we need to check if anything has changed (and load any new assets) during each p5 preload phase.

Reply:
👍